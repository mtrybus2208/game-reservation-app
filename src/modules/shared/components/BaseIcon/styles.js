import styled from 'styled-components';

export const BaseIcon = styled.img`
  display: flex;  
  margin: 0;
  padding: 0;
  width: ${({ size }) => size};
`;
