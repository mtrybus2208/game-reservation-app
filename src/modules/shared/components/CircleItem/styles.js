import styled, { css } from 'styled-components';
import { rgba } from 'polished';

export const CircleItem = styled.div`
    position: relative;
    border: solid 5px #23272d;
    box-shadow:
        0px 17px 24px 0 rgba(0, 0, 0, 0.33),
        inset 0px 2px 3px 0 rgba(255, 255, 255, 0.1);
    border: none;
    background-color: ${({ active }) => active ? '#bd790b' : '#23272d'};
    flex: 1 1 100%;
    border-radius: 100%;
    ${props => props.hoverable && css`
      &:hover {
        background: ${({ active }) => active ? '#bd790b' : rgba('#bd790b', 0.3)};
      }
    `}
`;

export const Body = styled.div`
    width: 85%;
    height: 85%;
    overflow: hidden;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%,-50%);
    box-shadow: -1px 1px 30px -3px rgba(0,0,0,0.21);
    border: none;
    background-color: #191b21;
    border-radius: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
`;
