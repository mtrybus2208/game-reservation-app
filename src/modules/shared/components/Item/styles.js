import styled from 'styled-components';

export const Item = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex: 1;
`;
