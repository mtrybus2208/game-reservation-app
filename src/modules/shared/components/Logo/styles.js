import styled from 'styled-components';

export const Logo = styled.div`
    display: flex;
    flex-direction: row;
    height: 35px;
`;

export const Image = styled.img`
    height: auto;
    box-sizing: content-box;
    width: 26px;
    padding: 0 21px;
    @media ${({ theme }) => theme.media.laptop} {
        width: 30px;
        padding: 0 16px 0 30px;
        border-right: 2px solid #6d6e6f;
    }
`;

export const Title = styled.div`
    margin-top: 3px;
    @media ${({ theme }) => theme.media.laptop} {
        margin-top: 0;
        margin-left: 18px;
    }
`;

export const Header = styled.div`
    font-family: Roboto;
    font-size: 13px;
    font-weight: normal;
    line-height: 1.19;
    letter-spacing: 2.4px;
    text-align: left;
    text-transform: uppercase;
    color: #6d6e6f;
    @media ${({ theme }) => theme.media.laptop} {
        font-size: 17px;
        line-height: 16px;
        letter-spacing: 3.15px;
    }
`;

export const HeaderHighlighted = styled.span`
    color: #8f6618;
`;

export const Description = styled.div`
    font-family: Roboto;
    font-size: 10px;
    font-weight: 300;
    line-height: 1.94;
    letter-spacing: .2px;
    text-align: left;
    color: #8f6618;
    @media ${({ theme }) => theme.media.laptop} {
        margin-top: 5px;
        font-size: 11px;
        line-height: 1.45;
        letter-spacing: .3px;
    }
`;