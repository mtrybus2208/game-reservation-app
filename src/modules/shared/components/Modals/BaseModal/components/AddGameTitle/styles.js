import styled from 'styled-components';

export const Bold = styled.span`
  color: ${props => props.theme.accent};
`;
