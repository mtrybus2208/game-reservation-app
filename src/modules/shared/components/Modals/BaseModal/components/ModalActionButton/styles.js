import styled from 'styled-components';
import { rgba } from 'polished';

export const ModalActionButton = styled.button`
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  background: none;
  border: none;
  padding: 15px;
  flex-grow: 1;
  height: 100%;
  outline-color: ${props => rgba(props.theme.accent, .2)};
`;
