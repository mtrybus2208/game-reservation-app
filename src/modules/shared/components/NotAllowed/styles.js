import styled from 'styled-components';

export const NotAllowed = styled.div`
  border-radius: 100%;
  width: 60px;
  height: 60px;
  display: flex;
  justify-content: center;
  width: 100%;
`;

export const Content = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

export const Info = styled.p`
  margin: 0;
  padding: 0;
  margin-top: 10px;
  font-size: ${({ size }) => size };
  color: ${({ fontColor }) => fontColor || 'inherit' }
`; 