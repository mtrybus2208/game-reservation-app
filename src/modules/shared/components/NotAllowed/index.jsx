import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import theme from '@/theme';
import  { ReactComponent as Alert }  from '../../../../assets/Icons/alert.svg';
import * as S from './styles';

const propTypes = {
  customIcon: PropTypes.node,
  msg: PropTypes.string,
  size: PropTypes.string,
  iconColor: PropTypes.string,
  fontColor: PropTypes.string,
};

const defaultProps = {
  msg: 'Not Allowed',
  size: 'sm',
  iconColor: theme.accent,
  fontColor: 'inherit',
};

const sizesObj = {
  sm: {
    width: '40px',
    font: '12px',
  },
  md: {
    width: '60px',
    font: '14px',
  },
  lg: {
    width: '80px',
    font: '16px',
  },
};

const NotAllowed = ({
  customIcon,
  msg,
  size,
  iconColor,
  fontColor,
}) => {
  const iconData = {
    fill: iconColor,
    width: sizesObj[size].width,
    height: sizesObj[size].width,
  };

  return (
    <S.NotAllowed>
      <S.Content>
        {
          customIcon
            ? React.cloneElement(customIcon, iconData)
            : <Alert {...iconData} />
        }

        <S.Info size={sizesObj[size].font} >
          {msg}
        </S.Info>
      </S.Content>
    </S.NotAllowed>
  )
};

NotAllowed.propTypes = propTypes;
NotAllowed.defaultProps = defaultProps;
export default NotAllowed;
