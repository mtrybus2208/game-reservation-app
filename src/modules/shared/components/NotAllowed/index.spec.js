import React, { useContext } from 'react';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import NotAllowed from './index';
import * as S from './styles';

describe('<NotAllowed />', () => {

  beforeEach(() => {});

  afterEach(() => {});
  it('renders without crashing', () => {
    shallow(<NotAllowed />);
  });

  it('should matches the snapshot', () => {
    const notAllowed = shallow(<NotAllowed />);

    expect(toJson(notAllowed)).toMatchSnapshot();
  }); 

  it('should render Alert icon from default props', () => {
    const notAllowed = shallow(<NotAllowed />);

    expect(notAllowed.find('icon-mock')).toHaveLength(1);
  });

  it('should render given icon from customIcon props', () => {
    const Mock = () => 'test';
    const notAllowed = shallow(<NotAllowed customIcon={<Mock />} />);

    expect(notAllowed.find(Mock)).toHaveLength(1);
  });
});
