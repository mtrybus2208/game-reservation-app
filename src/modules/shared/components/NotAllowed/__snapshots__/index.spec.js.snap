// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotAllowed /> should matches the snapshot 1`] = `
<styled.div>
  <styled.div>
    <icon-mock
      fill="#bd790b"
      height="40px"
      width="40px"
    />
    <styled.p
      size="12px"
    >
      Not Allowed
    </styled.p>
  </styled.div>
</styled.div>
`;
